gravity-forms/gw-unique-id-exclude-characters.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+* Gravity Wiz // Exclude characters from GPUID
+*
+* Allows you to exclude characters from unique IDs generated using GPUID.
+*
+* @version   1.2
+* @author    Eihab Ibrahim <eihab@gravitywiz.com>
+* @license   GPL-2.0+
+* @link      http://gravitywiz.com/calculate-number-of-days-between-two-dates/
+* @copyright 2020 Gravity Wiz
+*/
+class GWExcludeUIDChars {
+	/**
+	 * GWExcludeUIDChars Constructor
+	 *
+	 * Expects an $args named array with the following keys=>values:
+	 * 'form_id':  ID of the GF form to perform replacements in
+	 * 'field_id': GPUID Input ID
+	 * 'exclude':  String containing characters to exclude
+	 * 'sitewide': [optional] Boolean, set to true to apply to all GPUIDs
+	 *
+	 * Example usage in functions.php:
+	 * new GWExcludeUIDChars( array(
+	 *     'form_id'  => 11,
+	 *     'field_id' => 1,
+	 *     'exclude'  => 'li0o'
+	 * ) );
+	 */
+	public function __construct( $args ) {
+		extract( wp_parse_args( $args, array(
+			'form_id'  => false,
+			'field_id' => false,
+			'exclude'  => false,
+			'sitewide' => false,
+		) ) );
+		if ( ! $form_id || ! $field_id || ! $exclude ) {
+			if ( ! $sitewide || ! $exclude ) {
```

Comment:
Can this nested conditional be DRYed up any? `$exclude` appears to be repeated.

Reply:
The intention here was to allow for site-wide application or a form/field specific targeting. The second `$exclude` check is required just in case the first conditional fails on missing `$form_id` or `$field_id`.